### ISSUE TITLE
Live Component - Empty form data

### ISSUE THREAD

**Issue by sandoche** [symfony/ux#139]
Hi,

I am trying to make a Todo list with the live components.

I am using the form way described in the Live Component documentation.
The AJAX query runs when I click on the action button but it sends an empty `text` field.

- Here is my Task entity: https://github.com/sandoche/Symfony-ux-live-component-todo-list/blob/main/src/Entity/Task.php
- The create task component: https://github.com/sandoche/Symfony-ux-live-component-todo-list/blob/main/src/Components/CreateTaskForm.php
- The related view: https://github.com/sandoche/Symfony-ux-live-component-todo-list/blob/main/templates/components/create_task_form.html.twig
The full repo: https://github.com/sandoche/Symfony-ux-live-component-todo-list/

As you can see below the task['test'] is empty even though I filled the form:
![image](https://user-images.githubusercontent.com/3635348/134705089-d8ed7dbd-6ee5-41f6-ba06-2e31c1ea36c7.png)

Also it's not well documented I had to use `dehydrateWith` maybe I haven't done correctly.
I am not very sure if it's a bug or if I did something wrong.

**Comment by weaverryan** [comment]
Hey @sandoche!

Excellent reproducer! The fix you need is inside of your component's template - `create_task_form.html.twig`:

```diff
<div
	{{ init_live_component(this) }}
+	data-action="change->live#update"
>
```

When you trigger a "live action" (like you're doing when the form submits), the component "sends" all the data from the component. Buuuut, this does not blindly submit, for example, all of the fields inside the `form`. The flow needs to go like this:

1) The user types into the input
2) Something (this is the part that was missing) like a `change` event or `input` event uses the text in this box to update that piece of data inside the component.
3) You perform the action (e.g. submit the form).

I think that your mistake will likely be fairly common... but I'm not sure yet what we could do to be more clear and avoid it.

> Also it's not well documented I had to use dehydrateWith maybe I haven't done correctly.

You did a good job finding this :). What I've been doing so far - to avoid the problem of a non-persisted object from not being "dehydratable" - is to pass in `null` to the template... but it's not super smooth yet. To see what I'm talking about, you would:

A) Remove your `dehydrateWith`
B) Pass `null` as the `task` variable when rendering this component, something like:

```twig
{{ component('create_task_form', {
	task: this.task.id ? this.task : null,
	form: this.form
}) }}
```

This can be documented better - I'll open a PR!

Let me know what you think - and thanks for testing!

**Comment by sandoche** [comment]
It does work thanks a lot for your time @weaverryan! One question what is the dehydrateWith for? in which case should it be used?

**Comment by sandoche** [comment]
Also @weaverryan I have to other issues linked to this one:

1. When I add `data-poll` to the [`todo_list.html.twig`](https://github.com/sandoche/Symfony-ux-live-component-todo-list/blob/issue-data-poll/templates/components/todo_list.html.twig)
I get the following error:
```
Impossible to access an attribute ("id") on a null variable.
```
I managed to fix it using a workaround:
```twig
<div {{ init_live_component(this) }} data-poll>
	{% set task = null %}
	{{ component('create_task_form', {
        task: task,
        form: this.form
    }) }}


	{% for task in this.tasks %}
		<br>{{ task.text }}
	{% endfor %}
</div>
```

2. My second issue is that I don't find a way to update the parent view once the form is submited.
Basically thanks to your fix once the child form is submitted, the task is added to the database. But the parent is not reloading. I am not sure how to achieve this. I am using poll as a workaround for now.

**Comment by Kocal** [comment]
> Impossible to access an attribute ("id") on a null variable.

Change your code to: 

```diff
{{ component('create_task_form', {
-	task: this.task.id ? this.task : null,
+	task: this.task ? this.task.id : null,
	form: this.form
}) }}
```
